de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/model/interceptor/C_BPartner_CreditLimit.java

```diff
@@ -0,0 +1,54 @@
+package org.adempiere.bpartner.model.interceptor;
+
+import java.util.Collections;
+
+import org.adempiere.ad.modelvalidator.annotations.Interceptor;
+import org.adempiere.ad.modelvalidator.annotations.ModelChange;
+import org.adempiere.ad.trx.api.ITrx;
+import org.adempiere.bpartner.service.IBPartnerStatisticsUpdater;
+import org.adempiere.util.Services;
+import org.compiere.model.I_C_BPartner_CreditLimit;
+import org.compiere.model.ModelValidator;
+import org.compiere.util.Env;
+
+import lombok.NonNull;
+
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2017 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+@Interceptor(I_C_BPartner_CreditLimit.class)
+public class C_BPartner_CreditLimit
+{
+	public static final C_BPartner_CreditLimit INSTANCE = new C_BPartner_CreditLimit();
```

Comment:
annotate the class with `@Component`, remove the `INSTANCE` and the constructor.
Don't register it. It will automatically discovered.